python/ray/data/tests/test_backpressure_policies.py

```diff
@@ -138,6 +138,37 @@ def test_e2e_normal(self):
         start2, end2 = ray.get(actor.get_start_and_end_time_for_op.remote(2))
         assert start1 < start2 < end1 < end2, (start1, start2, end1, end2)
 
+    def test_e2e_time_backpressure(self):
+        """A simple E2E test with ConcurrencyCapBackpressurePolicy enabled."""
+        # TODO: merge this test with the above once we are exporting the
+        # backpressure_time in op_runtime_metrics.py. This test currently
+        # mutates the OpRuntimeMetrics dataclass to observe the backpressure time.
+        try:
+            from ray.data._internal.execution.interfaces.op_runtime_metrics import (
+                OpRuntimeMetrics,
+            )
+
+            OpRuntimeMetrics.__dataclass_fields__["backpressure_time"].metadata = {
+                "export_metric": True
+            }
+            actor = self._create_record_time_actor()
+            map_func1 = self._get_map_func(actor, 1)
+            map_func2 = self._get_map_func(actor, 2)
+
+            # Creat a dataset with 2 map ops. Each map op has N tasks, where N is
+            # the number of cluster CPUs.
+            N = self.__class__._cluster_cpus
+            ds = ray.data.range(N, parallelism=N)
+            # Use different `num_cpus` to make sure they don't fuse.
+            ds = ds.map_batches(map_func1, batch_size=None, num_cpus=1, concurrency=1)
+            ds = ds.map_batches(map_func2, batch_size=None, num_cpus=1.1, concurrency=1)
+            ds.take_all()
+            assert 0 < ds._plan.stats().extra_metrics["backpressure_time"] < 1
```

Comment:
maybe a short comment explaining how the upper limit 1 was calculated/estimated?

Reply:
Actually going to remove this upper limit, we re really just testing that this time is greater than 0. 1 was just chosen arbitrarily.